### ISSUE TITLE
Filtered Bulk Delete Ends Up Deleting Everything

### ISSUE THREAD

**Issue by nathanclevenger** [payloadcms/payload#9374]
### Describe the Bug

If a list view is searched, and then the `Select all x Resources` is clicked, and then if delete is selected, all documents from the collection are deleted not just the filtered/selected items.

This is an extremely serious issue - we just had to roll back a database to backup and lost data because of this bug.

### Link to the code that reproduces this issue

https://github.com/nathanclevenger/payload-repro

### Reproduction Steps

Go to a collection list view:
![Image](https://github.com/user-attachments/assets/93edafb1-882d-43f9-9a25-c13c6445e929)

Enter a search term:
![Image](https://github.com/user-attachments/assets/5151c4d2-ab1b-4a05-b69d-fb97954ecf34)

Hit the top checkbox to select all:
![Image](https://github.com/user-attachments/assets/88f080dd-0ac7-4942-9303-44b36ff5e925)

Hit the link to "Select All x Resources:
![Image](https://github.com/user-attachments/assets/324aa196-00fd-443f-b297-f46bcda9399b)

Hit Delete, and Confirm: (note, it shows the filtered count here)
![Image](https://github.com/user-attachments/assets/7689d84d-a202-49be-82c8-1e2073633987)

Then all resources in the collection will be deleted:
![Image](https://github.com/user-attachments/assets/a9059168-ad83-4b0b-83a2-13e9697174be)

I confirmed this exists in both Mongo and Postgres

### Which area(s) are affected? (Select all that apply)

Not sure

### Environment Info

```text
Payload: 3.0.1
Nextjs: 15.0.0
Node: 20.13.1
```

**Comment by jacobsfletch** [comment]
@nathanclevenger I'm unable to reproduce this within either our test suite or the reproduction repo you've provided. Here's a quick screen recording from my end using the steps you've outlined:

https://github.com/user-attachments/assets/ea36cd54-e653-4110-8dfc-61e075520d8b

**Comment by jacobsfletch** [comment]
Ahh @nathanclevenger missed a step, just coming back in here to confirm but I see @tyteen4a03 is too quick for me. Looking into it.

**Comment by jacobsfletch** [comment]
> Your setup has 3 and you select 1, while their setup has 4 and selected 3.

For the record it's the "Hit the link to "Select All x Resources" step that reproduces this issue. Not the number of documents.

**Comment by nathanclevenger** [comment]
> > Your setup has 3 and you select 1, while their setup has 4 and selected 3.
> 
> For the record it's the "Hit the link to "Select All x Resources" step that reproduces this issue. Not the number of documents.

Yes, it's the `Select All x Resources` that causes this issue ... I just confirmed it also affects Updates as well as Deletes
